framework/src/media/audio/audio_manager.c

```diff
@@ -159,18 +160,30 @@ static audio_manager_result_t set_audio_volume(audio_card_type_t type, uint8_t v
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
-static audio_manager_result_t find_audio_card_if_empty(audio_card_type_t card_type)
+static audio_manager_result_t init_audio_stream(audio_card_type_t card_type)
 {
-	int id = g_actual_audio_out_card_id;
-
-	if (card_type == INPUT) {
-		id = g_actual_audio_in_card_id;
+	if (((card_type == INPUT) && (g_actual_audio_in_card_id >= 0)) || ((card_type == OUTPUT) && (g_actual_audio_out_card_id >= 0))) {
+		goto select_card;
 	}
 
-	if ((id == INVALID_ID) && (find_audio_card(card_type) != AUDIO_MANAGER_SUCCESS)) {
-		meddbg("Found no active audio card type : %d\n", card_type);
+	/* check available cards and try to find all cards if there's no one */
+	if (find_audio_card(card_type) != AUDIO_MANAGER_SUCCESS) {
+		meddbg("Found no active %s audio card\n", (card_type == OUTPUT ? "output" : "input"));
```

Comment:
Init finding everything now, then no more need AUDIO_MANAGER_CARD_NOT_FOUND,  AUDIO_MANAGER_NO_AVAIL_CARD is enough, I think

Reply:
Right. Modified.